### ISSUE TITLE
Updated to Godot 4.4

### ISSUE THREAD

**Issue by TaraSophieDev** [MenacingMecha/godot-n64-shader-demo#6]
Will this repo be updated for Godot 4.4?

Vertex Shading/Lighting will be reimplemented.

Also the Godot Camera node got a new property called [Compositor](https://docs.godotengine.org/en/stable/tutorials/rendering/compositor.html) that seemingly makes it easier with viewport/post processing shaders. Instead of having a lot of viewports, the post processing shader can be implemented there.

Godot 4.3 also introduced [depth fog](https://github.com/godotengine/godot/pull/84792), like 3.x had.

**Comment by MenacingMecha** [comment]
Yes, this will be updated for 4.4 when it's fully released, it's an update like that I was waiting for.

Outside of a PR, will hold off for the time being on the depth fog.
